hooks/ useDisplayPreferences.tsx

```diff
@@ -0,0 +1,24 @@
+import { useRouter } from 'next/router';
+import { useEffect } from 'react';
+
+import { useLocalStorage } from './useLocalStorage';
+
+export type DisplayPreferences = 'list' | 'grid';
+
+export const useDisplayPreferences = () => {
+  const router = useRouter();
+  const [display, setDisplay] = useLocalStorage<DisplayPreferences>('display-preferences', 'list');
+
+  useEffect(() => {
+    if (!display) {
+      return;
+    }
+    if (!router.asPath.startsWith(`/${display}`)) {
+      const [, , ...rest] = router.pathname.split('/');
+      const path = `/${display}/${rest.join('/')}`;
+      void router.replace(path, undefined, { shallow: false });
```

Comment:
AFAIR nie trzeba mu mówić żeby nie było shallow, bo tak jest domyślnie

```suggestion
      void router.replace(path);
```